### PR TITLE
Implemented the ELK algorithm to fix positioning issues

### LINKED ISSUE #23: Implement subgroup positioning
Solve the issues with subgroup positioning by:
- Implementing the Elk algorithm
- Find another solution

### PR DESCRIPTION
**Pull request by JoXW100**
Fixes #23
- Still some issues when determining node size
See:
image

### CODE REVIEW

File: src/components/diagram/customNodes/diagramNodeGroup.js [hunk 1]
```diff
@@ -29,4 +32,4 @@ const DiagramNodeGroupSmall = ({ id, data }) => {
   )
 }
 
-export default DiagramNodeGroupSmall
+export default DiagramNodeGroup
```
Comment (thomssonkarl) [inline src/components/diagram/customNodes/diagramNodeGroup.js:35]: The smaller group that was removed or rather put into this component should just be the activity node.

File: src/components/diagram/diagram.js [hunk 2]
```diff
@@ -13,75 +13,27 @@ import '../../@types/@flowTypes'
...
-
   /**
    * Translates node type to a size
    * @param {DataNode} node
    * @returns {{ width: number, height: number }}
    */
   const processSize = (node) => {
     switch (node.type) {
+      case 'start':
+        return { width: 120, height: 100 }
```
Comment (thomssonkarl) [inline src/components/diagram/diagram.js:24]: I dont think the valuebox and startnode have to be this large, they seem to take a lot of unnecessary space of the diagram.

**Review by thomssonkarl — COMMENTED** [review]
Consider breaking out the processing functionality of diagram.js into a separate processing functions file. This file is starting to become big and unreadable.

File: src/components/diagram/customNodes/diagramNodeGroup.js (hunk 1 above)
Comment (JoXW100) [reply src/components/diagram/customNodes/diagramNodeGroup.js:35]: That will be resolved in Issue: #24

File: src/components/diagram/diagram.js (hunk 2 above)
Comment (JoXW100) [reply src/components/diagram/diagram.js:24]: the size is not only about the actual size of the component but also positioning, having a much smaller node with everything else looked strange. this value is temporary either way

**Review by thomssonkarl — APPROVED** [review]

**Review by mackan1410 — APPROVED** [review]

**Pull request merged**